pkg/health/health.go

```diff
@@ -96,12 +114,18 @@ func (s *ConsulHealthChecker) LookupHealth(serviceID string) (*ServiceStatus, er
 	serviceStatus := ServiceStatus{
 		Statuses: make(map[string]*ServiceNodeStatus),
 	}
+
+	var firstErr error = nil
 	for _, entry := range entries {
 		if serviceID == entry.Service.ID {
-			serviceStatus.Statuses[entry.Node.Node] = s.toNodeStatus(*entry)
+			status, err := s.toNodeStatus(serviceID, *entry)
+			if err != nil && firstErr == nil {
+				firstErr = err
```

Comment:
I'm still feeling the tingle of error suppression but this will at least signal that things are messed up.